src/service.ts

```diff
@@ -0,0 +1,250 @@
+import { Container, Contracts, Enums as AppEnums, Services, Utils as AppUtils } from "@arkecosystem/core-kernel";
+import { Interfaces } from "@arkecosystem/crypto";
+import axios from "axios";
+import os from "os";
+
+import { IOptions } from "./interface";
+import * as messages from "./messages";
+
+const VALID_EVENTS = [
+    "block.applied",
+    "block.forged",
+    "block.reverted",
+    "delegate.registered",
+    "delegate.resigned",
+    "forger.failed",
+    "forger.missing",
+    "forger.started",
+    "peer.added",
+    "peer.removed",
+    "transaction.applied",
+    "transaction.expired",
+    "transaction.forged",
+    "transaction.reverted",
+    "wallet.vote",
+    "wallet.unvote",
+    "round.created",
+    "activedelegateschanged",
+];
+
+const CUSTOM_EVENTS = ["activedelegateschanged"];
+const CUSTOM_EVENT_MAPPING = {
+    activedelegateschanged: AppEnums.BlockEvent.Forged, // AppEnums.RoundEvent.Created,
+};
+
+let LAST_ACTIVE_DELEGATES_CACHED: string[] = [];
+
+@Container.injectable()
+export default class Service {
+    @Container.inject(Container.Identifiers.EventDispatcherService)
+    private readonly emitter!: Contracts.Kernel.EventDispatcher;
+
+    @Container.inject(Container.Identifiers.LogService)
+    private readonly logger!: Contracts.Kernel.Logger;
+
+    @Container.inject(Container.Identifiers.TriggerService)
+    private readonly triggers!: Services.Triggers.Triggers;
+
+    // @Container.inject(Container.Identifiers.DposState)
+    // @Container.tagged("state", "clone")
+    // private readonly dposState!: Contracts.State.DposState;
+
+    @Container.inject(Container.Identifiers.WalletRepository)
+    // why state, blockchain - why not state, clone?
+    @Container.tagged("state", "blockchain")
+    private readonly walletRepository!: Contracts.State.WalletRepository;
+
+    private events = {};
+
+    public async listen(options: IOptions): Promise<void> {
+        LAST_ACTIVE_DELEGATES_CACHED = await this.getActiveDelegates();
+
+        for (const webhook of options.webhooks) {
+            for (const event of webhook.events) {
+                if (!VALID_EVENTS.includes(event)) {
+                    this.logger.warning(
+                        `[deadlock-delegate/notifier] ${event} is not a valid event. Check events in your deadlock-notifier configuration`,
+                    );
+                    continue;
+                }
+
+                if (!this.events[event]) {
+                    this.events[event] = [];
+                }
+
+                this.events[event].push({
+                    endpoint: webhook.endpoint,
+                    payload: webhook.payload,
+                });
+            }
+        }
+        Object.keys(this.events).forEach((event) => this.subscribe(event));
+    }
+
+    private async getActiveDelegates() {
+        // im not sure why I need to use triggers, but trying to inject dposState and calling
```

Comment:
Actions that are registered as triggers are intentionally not callable directly because there are `before` and `after` callbacks that could be registered by other plugins or parts of the kernel. These would fail to be executed if you call the action directly and could result in state corruption depending on what the action does.